modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -374,21 +374,17 @@ protected NodeChannels connectToChannels(DiscoveryNode node, ConnectionProfile p
         return nodeChannels;
     }
 
-    private class ChannelCloseListener implements ChannelFutureListener {
-
-        private final DiscoveryNode node;
-
-        private ChannelCloseListener(DiscoveryNode node) {
-            this.node = node;
-        }
-
+    private final class ChannelCloseListener implements ChannelFutureListener {
         @Override
         public void operationComplete(final ChannelFuture future) throws Exception {
-            onChannelClosed(future.channel());
-            NodeChannels nodeChannels = connectedNodes.get(node);
-            if (nodeChannels != null && nodeChannels.hasChannel(future.channel())) {
-                threadPool.generic().execute(() -> disconnectFromNode(node, future.channel(), "channel closed event"));
+            try {
+                onChannelClosed(future.channel());
```

Comment:
you are right, I simplified this by simply calling `disconnectFromNodeChannel` from `onChannelClosed` only. It's much simpler now but I think we can improve it further down the road.